engine/src/test/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionTest.java

```diff
@@ -61,6 +70,119 @@ public void shouldWriteDeletedEventsForMultipleDecisions() {
     verifyResourceIsDeleted(drgKey);
   }
 
+  @Test
+  public void shouldCreateIncidentIfOnlyDecisionVersionIsDeleted() {
+    // given
+    final long drgKey = deployDrg(DRG_SINGLE_DECISION);
+    final var processId = deployProcessWithBusinessRuleTask("jedi_or_sith");
+
+    // when
+    engine.resourceDeletion().withResourceKey(drgKey).delete();
+    final long processInstanceKey = engine.processInstance().ofBpmnProcessId(processId).create();
+
+    // then
+    assertThat(
+            RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .getFirst())
+        .describedAs("Should create incident when the only version of a decision is deleted")
+        .extracting(
+            r -> r.getValue().getBpmnProcessId(),
+            r -> r.getValue().getElementId(),
+            r -> r.getValue().getErrorType(),
+            r -> r.getValue().getErrorMessage())
+        .containsOnly(
+            processId,
+            "task",
+            ErrorType.CALLED_DECISION_ERROR,
+            """
+            Expected to evaluate decision 'jedi_or_sith', \
+            but no decision found for id 'jedi_or_sith'\
+            """);
+  }
+
+  @Test
+  public void shouldEvaluatePreviousDecisionVersionIfLatestVersionIsDeleted() {
+    // given
+    final long drgKeyV1 = deployDrg(DRG_SINGLE_DECISION);
+    final long drgKeyV2 = deployDrg(DRG_SINGLE_DECISION_V2);
+    final var processId = deployProcessWithBusinessRuleTask("jedi_or_sith");
+
+    // when
+    engine.resourceDeletion().withResourceKey(drgKeyV2).delete();
+    final long processInstanceKey =
+        engine
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withVariable("lightsaberColor", "blue")
+            .create();
+
+    // then
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limitToProcessInstanceCompleted())
+        .describedAs("Process Instance should be completed")
+        .extracting(r -> r.getValue().getBpmnElementType(), Record::getIntent)
+        .containsSubsequence(
+            tuple(BpmnElementType.BUSINESS_RULE_TASK, ProcessInstanceIntent.ELEMENT_COMPLETING),
+            tuple(BpmnElementType.BUSINESS_RULE_TASK, ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));
```

Comment:
🔧 It would be enough to make sure the decision is called, IMO this assertions is not needed